storage/src/mmr/verification.rs

```diff
@@ -1,24 +1,45 @@
 use crate::mmr::hasher::Hasher;
-use crate::mmr::iterator::PeakIterator;
+use crate::mmr::iterator::{PathIterator, PeakIterator};
+use crate::mmr::Error;
 use bytes::{Buf, BufMut};
 use commonware_cryptography::{Digest, Hasher as CHasher};
 
-/// A `Proof` contains the information necessary for proving the inclusion of an element, or some
-/// range of elements, in the MMR from its root hash. The `hashes` vector contains: (1) the peak
-/// hashes other than those belonging to trees containing some elements within the range being
-/// proven, followed by: (2) the nodes in the remaining perfect trees necessary for reconstructing
-/// their peak hashes from the elements within the range. Both segments are ordered by decreasing
-/// height.
+/// Contains the information necessary for proving the inclusion of an element, or some range of
+/// elements, in the MMR from its root hash.
+///
+/// The `hashes` vector contains: (1) the peak hashes other than those belonging to trees containing
+/// some elements within the range being proven, followed by: (2) the nodes in the remaining perfect
+/// trees necessary for reconstructing their peak hashes from the elements within the range. Both
+/// segments are ordered by decreasing height.
 #[derive(Clone, Debug, Eq)]
-/// A Proof contains the information necessary for proving the inclusion of an element, or some
-/// range of elements, in the MMR.
 pub struct Proof<H: CHasher> {
     /// The total number of nodes in the MMR.
     pub size: u64,
-    /// The hashes necessary for proving the inclusion of an element, or range of elements, in the MMR.
+    /// The hashes necessary for proving the inclusion of an element, or range of elements, in the
+    /// MMR.
     pub hashes: Vec<H::Digest>,
 }
 
+// A trait for MMRs that can generate inclusion proofs.
+pub trait VerifiableMmr<H: CHasher> {
```

Comment:
I wonder if you should break this into `Storage` and `Verifier` (where `Verifier` isn't a trait?)

Reply:
I need a trait though (as far as I understand) so I can make the verification functions generic of the implementation (whether in-mem vs journaled)

Reply:
I simplified the trait to contain just the size & get_node method.